sklearn/preprocessing/_encoders.py

```diff
@@ -631,11 +631,12 @@ class OrdinalEncoder(_BaseEncoder):
 
         .. versionadded:: 0.24
 
-    unknown_value : int, default=None
+    unknown_value : int or np.nan, default=None
         When the parameter handle_unknown is set to 'use_encoded_value', this
         parameter is required and will set the encoded value of unknown
         categories. It has to be distinct from the values used to encode any of
-        the categories in `fit`.
+        the categories in `fit`. If set to np.nan, the `dtype` parameter must
+        be either a float dtype or `object`.
```

Comment:
I am thinking that `dtype` was used for `np.float64` or `np.float32` and not object. If it was object, there isn't a good point in running `OrdinalEncoder`?

Reply:
I'm not sure, we do have tests about `object` dtype. Also I would assume some users use the OE to encode categorical features for trees, and they might be strings?

Reply:
I do not see any test that sets the `OrdinalEncoder` dtype parameter to object.

For this PR, may we assume that `dtype` is numerical?

Reply:
Right my bad, I was mixing input and output. We now raise whenever the dtype isn't a float dtype if unknown_value=nan